staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go

```diff
@@ -299,11 +299,26 @@ func (le *LeaderElector) release() bool {
 		RenewTime:            now,
 		AcquireTime:          now,
 	}
-	if err := le.config.Lock.Update(context.TODO(), leaderElectionRecord); err != nil {
-		klog.Errorf("Failed to release lock: %v", err)
+	todoCtx := context.TODO()
+	// update the lock itself
+	if err := le.config.Lock.Update(todoCtx, leaderElectionRecord); err == nil {
```

Comment:
Commented with @wojtek-t offline and maybe we can use a Patch call directly, it may solve this duplication of code, let me pull @liggitt for this opinion too